lib/sles4sap.pm

```diff
@@ -72,6 +75,98 @@ sub become_sapadm {
+    return (defined $ret and $ret == 0);
+}
+
+sub is_nw_profile {
+    my $list = script_output "tuned-adm list";
+    return ($list =~ /sap-netweaver/);
+}
+
+sub prepare_profile {
+    my ($self, $profile) = @_;
+    return unless ($profile eq 'HANA' or $profile eq 'NETWEAVER');
+
+    # Will prepare system with saptune only if it's available.
+    # Otherwise will try to use the tuned 'sap-netweaver' profile
+    # for netweaver and the recommended one for hana
+    my $has_saptune = $self->is_saptune_installed();
+
+    if ($has_saptune) {
+        assert_script_run "tuned-adm profile saptune";
+        assert_script_run "saptune solution apply $profile";
+    }
+    elsif ($profile eq 'NETWEAVER') {
+        $profile = $self->is_nw_profile() ? 'sap-netweaver' : '$(tuned-adm recommend)';
+        assert_script_run "tuned-adm profile $profile";
+    }
+    elsif ($profile eq 'HANA') {
+        assert_script_run 'tuned-adm profile $(tuned-adm recommend)';
+    }
+
+    # Restart systemd-logind to ensure that all new connections will have the
+    # SAP tuning activated
+    systemctl 'restart systemd-logind.service';
+
+    # 'systemctl restart systemd-logind' is causing the X11 console to move
+    # out of tty2 on SLES4SAP-15, which in turn is causing the change back to
+    # the previous console in post_run_hook() to fail when running on systems
+    # with DESKTOP=gnome, which is a false positive as the test has already
+    # finished by that step. The following prevents post_run_hook from attempting
+    # to return to the console that was set before this test started. For more
+    # info on why X is running in tty2 on SLES4SAP-15, see bsc#1054782
+    $prev_console = undef;
+
+    # If running in DESKTOP=gnome, systemd-logind restart may cause the graphical console to
+    # reset and appear in SUD, so need to select 'root-console' again
+    assert_screen(
+        [
+            qw(root-console displaymanager displaymanager-password-prompt generic-desktop
+              text-login linux-login started-x-displaymanager-info)
+        ], 120);
+    $self->select_serial_terminal unless (match_has_tag 'root-console');
+
+    if ($has_saptune) {
+        assert_script_run "saptune daemon start";
+        assert_script_run "saptune solution verify $profile";
+        my $output = script_output "saptune daemon status";
+        record_info("tuned status", $output);
+    }
+    else {
+        assert_script_run "systemctl restart tuned";
+    }
+
+    my $output = script_output "tuned-adm active";
+    record_info("tuned profile", $output);
+}
+
+sub copy_media {
+    my ($self, $proto, $path, $nettout, $target) = @_;
+
+    # First copy media
+    assert_script_run "mkdir $target";
+    assert_script_run "mount -t $proto $path /mnt";
+    type_string "cd /mnt\n";
+    type_string "cd " . get_var('ARCH') . "\n";    # Change to ARCH specific subdir if exists
+    assert_script_run "tar -cf - . | (cd $target/; tar -pxf - )", $nettout;
```

Comment:
Did you tried `cp -ax`? IMHO it could be better because it avoids the use of a pipe (I had bad things sometimes with pipe...). I did some tests on my lab and `cp` is sometimes quicker (but I have to admit that the differences is very small).

Reply:
No. I was actually thinking of changing that command to a `rsync` but decided not to touch it as it's currently working in the NetWeaver installation tests. Let me try to change it a see how it works.

Reply:
Tried with `cp -ax` and with `rsync`. I got 23 minutes with the tar pipeline, time out with rsync and also 23 minutes with `cp -ax`. Will push a commit changing the tar pipeline to `cp -ax` as it's easier to read. Verification runs:

NW: http://mango.suse.de/tests/1000
Hana: http://mango.suse.de/tests/1001